docs/standard/commandline/syntax.md

```diff
@@ -0,0 +1,577 @@
+
+For some Windows command-line apps, you identify an option by using a leading slash (`/`) with the option name. For example:
+
+```console
+msbuild /version
+        ^------^
+```
+
+`System.CommandLine` supports both POSIX and Windows prefix conventions. When you [configure an option](define-commands.md#define-options), you specify the option name including the prefix.
+
+## Arguments
+
+An argument is a value passed to an option or a command. The following examples show an argument for the `verbosity` option and an argument for the `build` command.
+
+```console
+dotnet tool update dotnet-suggest --verbosity quiet --global
+                                              ^---^
+```
+
+```console
+dotnet build myapp.csproj
+             ^----------^
+```
+
+Arguments can have default values that apply if no argument is explicitly provided. For example, many options are implicitly Boolean parameters with a default of `true` when the option name is in the command line. The following command-line examples are equivalent:
+
+```dotnetcli
+dotnet tool update dotnet-suggest --global
+                                  ^------^
+
+dotnet tool update dotnet-suggest --global true
+                                  ^-----------^
+```
+
+Some options have required arguments. For example in the .NET CLI, `--output` requires a folder name argument. If the argument is not provided, the command fails.
+
+Arguments can have expected types, and the app errors if the argument can't be parsed into the expected type. For example, the following command errors because "silent" isn't one of the valid values for `--verbosity`:
+
+```dotnetcli
+dotnet build --verbosity silent
+```
+
+```output
+Cannot parse argument 'silent' for option '-v' as expected type 'Microsoft.DotNet.Cli.VerbosityOptions'. Did you mean one of the following?
+Detailed
+Diagnostic
+Minimal
+Normal
+Quiet
+```
+
+Arguments also have expectations about how many values can be provided. Examples are provided in the [section on argument arity](#argument-arity).
+
+## Order of options and arguments
+
+You can provide options before arguments or arguments before options on the command line. The following commands are equivalent:
+
+```dotnetcli
+dotnet add package System.CommandLine --prerelease
+dotnet add package --prerelease System.CommandLine 
+```
+
+Options can be specified in any order. The following commands are equivalent:
+
+```dotnetcli
+dotnet add package System.CommandLine --prerelease --no-restore --source https://api.nuget.org/v3/index.json
+dotnet add package System.CommandLine --source https://api.nuget.org/v3/index.json --no-restore --prerelease 
+```
+
+When there are multiple arguments, the order does matter. The following commands are not equivalent:
```

Comment:
Well, they might be. Maybe 

These pass a list with the same values to the handler, but they differ in the order of the values.